### PR TITLE
Feature/tsn 958 add collaborator modal

### PR DESCRIPTION
**Pull request by tmassuia-sw**

### CODE REVIEW

File: packages/apps/web/src/modals/MemoryAddCollaboratorModal/MemoryAddCollaboratorModal.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		}
+		);
+	});
+	const [mutation, mutationResult] = useMutation(UpsertMemoryMutation);
+
+	const UpsertMemory = () => {
+		let collaboratorsId = [];
+		currentCollaborators.forEach(e => {
+			collaboratorsId.push(parseInt(e.id));
+		});
```
Comment (frey-sw) [inline packages/apps/web/src/modals/MemoryAddCollaboratorModal/MemoryAddCollaboratorModal.js]: Can be simplified by:
const collaboratorsId = currentCollaborators.map( c => parseInt(c.id) )

File: packages/apps/web/src/modals/MemoryAddCollaboratorModal/MemoryAddCollaboratorModal.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+		let memoryStoriesId = [];
+		props.memory.stories.forEach(e => memoryStoriesId.push(parseInt(e.id)));
```
Comment (frey-sw) [inline packages/apps/web/src/modals/MemoryAddCollaboratorModal/MemoryAddCollaboratorModal.js]: Same as previous comment

File: packages/apps/web/src/modals/MemoryAddCollaboratorModal/MemoryAddCollaboratorModal.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+		mutation({
+			variables: {
+				input: {
+					id: parseInt(props.memory.id),
+					profileId: parseInt(profileContext.profile.id),
+					title: props.memory.title,
+					description: props.memory.description,
+					collaborators: collaboratorsId,
+					stories: memoryStoriesId
```
Comment (frey-sw) [inline packages/apps/web/src/modals/MemoryAddCollaboratorModal/MemoryAddCollaboratorModal.js]: Both improvements can be added each on a single line here

File: packages/apps/web/src/modals/MemoryAddCollaboratorModal/MemoryAddCollaboratorModal.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+					</h2>
+				</div>
+				{mutationResult.loading ? <Overlay type={OVERLAY_TYPES.LOADING} /> :
+					<>
+						<div className={styles.memoryInputCont}>
+							<ElementSelector
+								isProfileList 
+								selectedList={currentCollaborators}
+								elementList={relationsArray}
+								memoryOwner={props.memory.owner.id}
```
Comment (frey-sw) [inline packages/apps/web/src/modals/MemoryAddCollaboratorModal/MemoryAddCollaboratorModal.js]: This is not a generic UI element if it involves custom properties for functionalities

File: packages/ui-components/src/ElementSelector/ElementSelector.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+		</>
+	);
+};
+
+ElementSelector.propTypes = {
+	isProfileList: PropTypes.bool,
+	elementList: PropTypes.array.isRequired,
+	selectedList: PropTypes.array.isRequired,
+	updateItemCallBack: PropTypes.func.isRequired,
+	memoryOwner: PropTypes.number
```
Comment (frey-sw) [inline packages/ui-components/src/ElementSelector/ElementSelector.js]: This is not a generic UI element if it involves custom properties for functionalities

File: packages/ui-components/src/ElementSelector/ElementSelector.js [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+						<FontAwesomeIcon icon={faInfoCircle} className='marginRight_1' />
+						{t('UI.NO_ELEMENTS')}
+					</p>
+			}
+		</>
+	);
+};
+
+ElementSelector.propTypes = {
+	isProfileList: PropTypes.bool,
```
Comment (frey-sw) [inline packages/ui-components/src/ElementSelector/ElementSelector.js]: This is not a generic UI element if it involves custom properties for functionalities

File: packages/apps/web/src/modals/MemoryAddCollaboratorModal/MemoryAddCollaboratorModal.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+					<h2 className={styles.memoryTitle}>
+						<FontAwesomeIcon icon={faUserPlus} className={styles.memoryTitle__icon} />
+						{t('MEMORIES.ADD_COLLABORATOR_MODAL.TITLE_LABEL')}
+					</h2>
+				</div>
+				{mutationResult.loading ? <Overlay type={OVERLAY_TYPES.LOADING} /> :
+					<>
+						<div className={styles.memoryInputCont}>
+							<ElementSelector
+								isProfileList
```
Comment (frey-sw) [inline packages/apps/web/src/modals/MemoryAddCollaboratorModal/MemoryAddCollaboratorModal.js]: This is not a generic UI element if it involves custom properties for functionalities

File: packages/ui-components/src/ElementSelector/ElementSelector.js [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+import ProfileImage from '../ProfileImage/ProfileImage';
+import FlexContainer from '../FlexContainer/FlexContainer';
+
+
+const ElementSelector = (props) => {
+	const { t } = useTranslation();
+
+	const [state, setState] = useState({
+		searchText: '',
+		elementList: props.elementList.relations || [],
```
Comment (frey-sw) [inline packages/ui-components/src/ElementSelector/ElementSelector.js]: The .relations reference should be sent from the parent node

File: packages/apps/web/src/modals/MemoryAddCollaboratorModal/MemoryAddCollaboratorModal.js [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+						{t('MEMORIES.ADD_COLLABORATOR_MODAL.TITLE_LABEL')}
+					</h2>
+				</div>
+				{mutationResult.loading ? <Overlay type={OVERLAY_TYPES.LOADING} /> :
+					<>
+						<div className={styles.memoryInputCont}>
+							<ElementSelector
+								isProfileList 
+								selectedList={currentCollaborators}
+								elementList={relationsArray}
```
Comment (frey-sw) [inline packages/apps/web/src/modals/MemoryAddCollaboratorModal/MemoryAddCollaboratorModal.js]: The .relations reference should be sent from the parent node. Check ElementList.js

File: packages/ui-components/src/ElementSelector/ElementSelector.js [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+			return setState({ ...state, selectedList: copy });
+		}
+	};
+
+	props.updateItemCallBack(state.selectedList);
+
+	const filterElements = (el) => {
+		if (props.isProfileList) {
+			return props.elementList.filter(el => el.firstName.toLowerCase().includes(state.searchText.toLowerCase()));
+		}
```
Comment (frey-sw) [inline packages/ui-components/src/ElementSelector/ElementSelector.js]: To leave this filter, the component should recieve it by prop not using a custom prop

File: packages/ui-components/src/ElementSelector/ElementSelector.js [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+			</>
+		);
+	};
+
+	return (
+		<>
+			<Input
+				type={INPUT_TYPES.TEXT}
+				autofocus
+				placeholder={props.isProfileList ? t('MEMORIES.ADD_COLLABORATOR_MODAL.SEARCH_LABEL') : t('UI.SEARCH')}
```
Comment (frey-sw) [inline packages/ui-components/src/ElementSelector/ElementSelector.js]: This prop should not be used

File: packages/ui-components/src/ElementSelector/ElementSelector.js [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+				clearHandler={clearSearchHandler}
+			/>
+
+
+			{
+				state.selectedList && state.selectedList.length ?
+					<div className={styles.selectedListItems}>
+						{
+							state.selectedList.map(e => {
+								if (e.id !== props.memoryOwner) {
```
Comment (frey-sw) [inline packages/ui-components/src/ElementSelector/ElementSelector.js]: This filter should be done on the parent node of the generic ui element

File: packages/apps/web/src/modals/MemoryAddCollaboratorModal/MemoryAddCollaboratorModal.js (hunk 9 above)
Comment (tmassuia-sw) [reply packages/apps/web/src/modals/MemoryAddCollaboratorModal/MemoryAddCollaboratorModal.js]: Fixed on Element.js

**Pull request merged**